hipogram/posts/views.py

```diff
@@ -1,14 +1,24 @@
-from django.views.generic import ListView
-from django.shortcuts import redirect, render, get_object_or_404
-from django.contrib.auth.decorators import login_required
-from django.contrib import messages
+from django.views.generic import ListView, CreateView, DeleteView, UpdateView, View
+from django.contrib.auth.mixins import LoginRequiredMixin
+# from django.contrib import messages
 from django.db.models import Count
 from django.utils import timezone
 from django.http import HttpResponseRedirect
+from django.urls import reverse_lazy
 
 
 from .models import Post, Tag, Like, Rate
-from .forms import PostForm, RatePostForm
+from .forms import RatePostForm
+
+
+class OwnerRequiredMixin(LoginRequiredMixin):
+
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        post = self.get_object()
+        if post.created_by != self.request.user:
+            return self.handle_no_permission()
+        return response
```

Comment:
Looks nice :ok_hand: . We can try a little more to make it even more cleaner together in the meeting.

Reply:
I think it should be better on `core/mixins.py`

Reply:
I created one but now it doesn't work. I believe I have forgotten about something. Can you check?